ert/data/record/_record.py

```diff
@@ -59,19 +59,51 @@ class RecordType(str, Enum):
 
 
 class Record(ABC):
+    """The :py:class:`Record` class is an abstract class that
+    includes record data and the record type.
+
+    The record needs be either `BlobRecord` or `NumericalRecord` type derived
+    from `Record` class
+    """
+
     @property
     @abstractmethod
     def data(self) -> record_data:
+        """Returns record data.
+
+        Returns:
+            record_data: record data as
+            Union[numerical_record_data, blob_record_data]
+        """
         pass
 
     @property
     @abstractmethod
     def record_type(self) -> RecordType:
+        """Returns the type of the record data.
+
+        Returns:
+            RecordType: type of data
+        """
         pass
 
 
 class BlobRecord(Record):
+    """The :py:class:`BlobRecord` class is realization of the Record class
+    that handles binary data.
+    """
+
     def __init__(self, data: blob_record_data) -> None:
+        """Creates a BlobRecord instance from the binary record data.
+        It automatically assigns :py:class:`RecordType` as `RecordType.BYTES`
+
+        Args:
+            data: blob record data
+
+        Raises:
+            RecordValidationError: Raises when data is not binary failing
```

Comment:
Something odd with the sentence, has it been rewritten? "Raised when the data has failed the data validation, i.e. not binary".

Reply:
Should be fixed now